src/Sarif.Viewer.VisualStudio.UnitTests/Sarif/StackFrameExtensionsTests.cs

```diff
@@ -30,70 +27,123 @@ public void StackFrameExtensions_HasNoLocation_ShouldNotThrowException()
             };
 
             var stackFrameModel = stackFrame.ToStackFrameModel(resultId, runIndex);
-            stackFrameModel.Should().NotBeNull();
-            stackFrameModel.ResultId.Should().Be(resultId);
-            stackFrameModel.RunIndex.Should().Be(runIndex);
-            stackFrameModel.Module.Should().BeEquivalentTo(moduleName);
-            stackFrameModel.Message.Should().BeNull();
-            stackFrameModel.FullyQualifiedLogicalName.Should().BeNull();
-            stackFrameModel.FilePath.Should().BeNull();
-            stackFrameModel.Region.Should().BeNull();
-            stackFrameModel.Address.Should().Be(0);
-            stackFrameModel.Line.Should().Be(0);
-            stackFrameModel.Column.Should().Be(0);
-            stackFrameModel.Offset.Should().Be(0);
+
+            VerifyStackFrame(stackFrameModel, stackFrame, resultId, runIndex);
         }
 
         [Fact]
         public void StackFrameExtensions_HasLocation()
         {
             int resultId = 1;
             int runIndex = 0;
+            string stackFrameText = "stack frame location 1";
+            string fullyQualifiedName = @"namespaceA::namespaceB::classC";
```

Comment:
>fullyQualifiedName [](http://example.com/codeflow?start=19&length=18)

question, this fqdn looks very different from before change, is it by design

Reply:
ye its a fully qualifed name for LogicalLocation, the value can point to a logical location in source code like cs/cpp, or soruce file like json/xml. please see the [spec](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317724)